python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -207,7 +207,15 @@ def put(self, mine_incident_guid):
         for key, value in data.items():
             if key == 'dangerous_occurrence_subparagraph_ids':
                 continue
-            setattr(incident, key, value)
+            if key in [
+                    'reported_to_inspector_party_guid', 'responsible_inspector_party_guid',
+                    'determination_inspector_party_guid'
+            ]:
+                tmp_party = Party.query.filter_by(party_guid=value).first()
+                if tmp_party and 'INS' in tmp_party.business_roles_codes:
+                    setattr(incident, key, data.get(key))
```

Comment:
this should be `value` not `data.get(key)` my b.

Reply:
I'll change this on mine, no worries.